test/liquidityConversionRate.js

```diff
@@ -0,0 +1,719 @@
+        result =  await liqConvRatesInst.getRateWithE(token.address,false,qtyInSrcWei,eInFp);
+        assert.equal(result, 0, "bad result");
+
+        currentMaxSellRateInPrecision = maxSellRateInPrecision
+        currentMinSellRateInPrecision = gotResult.plus(100)
+        await liqConvRatesInst.setLiquidityParams(rInFp, pMinInFp, formulaPrecisionBits, maxCapBuyInWei, maxCapSellInWei, feeInBps, currentMaxSellRateInPrecision, currentMinSellRateInPrecision)
+        result =  await liqConvRatesInst.getRateWithE(token.address,false,qtyInSrcWei,eInFp);
+        assert.equal(result, 0, "bad result");
+
+        //return things to normal
+        await liqConvRatesInst.setLiquidityParams(rInFp, pMinInFp, formulaPrecisionBits, maxCapBuyInWei, maxCapSellInWei, feeInBps, maxSellRateInPrecision, minSellRateInPrecision)
+    });
+
+    it("should test exceeding max cap buy", async function () {
+        let qtyInSrcWei = BigNumber(maxCapBuyInEth + 0.2).mul(precision)
+        let result =  await liqConvRatesInst.getRateWithE(token.address,true,qtyInSrcWei,eInFp);
+        assert.equal(result, 0, "bad result");
+    });
+
+    it("should test exceeding max cap sell", async function () {
+        let sellQtyInTokens = (maxCapSellInEth / p0) * (1.1);
+        let sellQtyInTwi = BigNumber(sellQtyInTokens).mul(tokenPrecision)
+        let result =  await liqConvRatesInst.getRateWithE(token.address,false,sellQtyInTwi,eInFp);
+        assert.equal(result.valueOf(), 0, "bad result");
+    });
+
+    it("should test get rate", async function () {
+        let balance = await Helper.getBalancePromise(reserveAddress);
+        let eInEth = balance.div(precision)
+        let expectedResult = priceForDeltaE(feePercent, r, pMin, deltaE, eInEth).mul(precision)
+        let qtyInSrcWei = BigNumber(deltaE).mul(precision)
+        let result =  await liqConvRatesInst.getRate(token.address, 0, true, qtyInSrcWei);
+        assertAbsDiff(result, expectedResult, expectedDiffInPct)
+    });
+
+    it("should test recording of imbalance from non reserve address.", async function () {
+        let buyAmountInTwei = BigNumber(10).pow(tokenDecimals).mul(deltaT)
+        let expectedValueBeforeReducingFee = buyAmountInTwei.mul(3) / ((100 - feePercent)/100) // TODO - this calc is a duplication, move to general place...
+        let expectedResult = BigNumber(((feePercent / 100) * expectedValueBeforeReducingFee).toString())
+        try {
+            await liqConvRatesInst.recordImbalance(token.address, buyAmountInTwei, 3000, 3000, {from: operator})
+            assert(false, "expected to throw error in line above.")
+        }
+        catch(e){
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }    
+    });
+});
+
+
+contract('KyberReserve', function(accounts) {
+    it("should init globals. init ConversionRates Inst, token, set liquidity params .", async function () {
```

Comment:
better have a name that is different then kyberReserve because we already have one test for kyber reserve.
so if all regression is run we can know where it failed..
maybe write kyberReserve for Liquidity

Reply:
done. kyberReserve for Liquidity.